src/frontend/src/hooks/ProjectHooks.ts

```diff
@@ -0,0 +1,51 @@
+import { useState, useEffect } from 'react';
+import { IProject } from 'types/types';
+import ApiConfig from 'config/ApiConfig';
+import defaultRequestOptions from './defaultRequestHeaders';
+
+const useProjects = () => {
+    const [projects, setProjects] = useState<IProject[]>([]);
```

Comment:
I haven't added the loading flag here only because I didn't hit the race condition. That doesn't mean it's not happening though.